README.md

```diff
@@ -96,20 +96,49 @@ to your keystore like this:
 keytool -import -alias conjur-youraccount -keystore "$JRE_HOME/lib/security/cacerts"  -file ./conjur-youraccount.der
 ```
 
-## Basic Usage
+## Examples
 
-### Creating a Conjur Instance
+### Authorization Patterns
+All authorization options require the environment variables `CONJUR_ACCOUNT` and `CONJUR_APPLIANCE_URL` are set:
+```sh
+export CONJUR_ACCOUNT=<account specified during Conjur setup>
+export CONJUR_APPLIANCE_URL=<Conjur HTTPS endpoint>
+```
 
 A `Conjur` instance provides access to the individual Conjur services. To create one, you'll need the environment
 variables as described above. You will typically create a Conjur instance from these values in the following way:
 
+#### Environment Variables
+```sh
+# Additionally set the following environment variables:
+export CONJUR_AUTHN_LOGIN=<user/host identity>
+export CONJUR_AUTHN_API_KEY=<user/host API key>
+```
 ```java
+// Using environment variables
 Conjur conjur = new Conjur();
+```
 
+#### Username and Password
+```java
+// Authenticate using provided username/hostname and password/API key
+Conjur conjur = new Conjur('host/host-id', 'api-key');
+Conjur conjur = new Conjur('username', 'password');
 ```
 
-where the Conjur object is logged in to the account & ready for use.
+#### Credentials
+```java
+// Authenticate using a Credentials object
+Credentials credentials = new Credentials('username', 'password');
+Conjur conjur = new Conjur(credentials);
+```
 
+#### Authorization Token
+```java
+String authTokenString = new String(Files.readAllBytes(Paths.get('path/to/conjur/authentication/token.json')));
```

Comment:
I think that this should be encapsulated inside the library and not given by the developer. So the developer will only state that he/she is using a token from a file. The file should be given in the CONJUR_AUTHN_TOKEN_FILE environment variable or in the Conjur class constructor (instead of getting a token).
Once this is encapsulated, we need to also add a wait mechanism, in case the application is trying to use the token file before it got created by the sidecar. See similar mechanism here: https://github.com/cyberark/conjur-api-go/blob/b75a31bafc95e06780b81df18ba5dba8dd0dc32f/conjurapi/authn/token_file_authenticator.go#L23